lib/active_record/connection_adapters/postgis/schema_statements.rb

```diff
@@ -96,8 +96,9 @@ def initialize_type_map(map = type_map)
             st_point
             st_polygon
           ).each do |geo_type|
-            map.register_type(geo_type) do |oid, _, sql_type|
-              OID::Spatial.new(oid, sql_type)
+            map.register_type(geo_type) do |_, _, sql_type|
+              geo_type, srid, has_z, has_m, geographic = OID::Spatial.parse_sql_type(sql_type)
+              OID::Spatial.new(geo_type: geo_type, srid: srid, has_z: has_z, has_m: has_m, geographic: geographic)
```

Comment:
Since I don't know much about this adapter the question may sound stupid. Is `OID::Spatial` a private API only meant to be used by the adapter ? If not this is a major version we'll need to bump

Reply:
It should be a private API only used by the adapter. It defines how we parse data from a column with a SQL definition like `geometry(POINT, 4326)`. Now we're adding the ability to define those same properties with the attributes API. Typically attributes and columns will use the same `OID` type class to keep the type casting consistent for both. 

Here's the attributes registered in the postgres adapter (https://github.com/rails/rails/blob/ee7cf8cf7569ef87079c48ee81c867eae5e24ed4/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L948-L967)

Here's the database types registered in the postgres adapter (https://github.com/rails/rails/blob/ee7cf8cf7569ef87079c48ee81c867eae5e24ed4/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L517-L585).

Though if we didn't want to worry about breaking changes, I could essentially copy the `OID::Spatial` class to an `OID::SpatialAttribute` class and have that handle the attributes only.

Reply:
If this is all internal there is nothing to worry about, at most we can make it a minor..

Thanks for the explanation, it is much more clear !